File: BookNestBackend/BookNest.Model/Responses/LoginResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BookNest.Model.Responses
{
    public class LoginResponse
    {
        public int UserId { get; set; }
        public string Username { get; set; } = string.Empty;
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string EmailAddress { get; set; } = string.Empty;
        public List<string> Roles { get; set; } = new List<string>();
        public string Token { get; set; } = string.Empty;
        public DateTime ExpiresAt { get; set; }

    }
}

File: BookNestFrontend/booknest_desktop/lib/models/login_response.dart
class LoginResponse {
  final int userId;
  final String username;
  final String firstName;
  final String lastName;
  final String emailAddress;
  final List<String> roles;
  final String token;
  final DateTime expiresAt;

  LoginResponse({
    required this.userId,
    required this.username,
    required this.firstName,
    required this.lastName,
    required this.emailAddress,
    required this.roles,
    required this.token,
    required this.expiresAt,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      userId: json['userId'],
      username: json['username'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      emailAddress: json['emailAddress'],
      roles: List<String>.from(json['roles']),
      token: json['token'],
      expiresAt: DateTime.parse(json['expiresAt']),
    );
  }
}
